### PR TITLE
fix: compatible with latest dart version

### LINKED ISSUE #150: Support for intl 0.19.0 to be able to use the latest dart version
**Proposal:**
Short summary of the feature.

**Current behavior:**
Describe what currently happens.

**Desired behavior:**
Describe what you want.

**Alternatives considered:**
Describe other solutions or features you considered.

**Use case:**
Why is this important (helps with prioritizing requests)?

### PR DESCRIPTION
**Pull request by bednar**
Closes #148
Closes #149
Closes #150

## Proposed Changes

1. Removed `intl `as a required dependency — the `intl` package was used in the OpenApi generator, but our code does not utilize it.
2. `pubspec.lock` is not part of the repository because it determines library versions; thus, different Dart versions in CI may require different versions of dependencies.
3. The pub ecosystem in `dependabot` is no longer in beta status.

## Checklist

- [x] CHANGELOG.md updated
- [x] Rebased/mergeable
- [x] `pub run test` completes successfully
- [x] Commit messages are in [semantic format](https://seesparkbox.com/foundry/semantic_commit_messages)
- [x] Sign [CLA](https://www.influxdata.com/legal/cla/) (if not already signed)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,7 +6,7 @@
 .packages
 build/
 # If you're building an application, you may want to check-in your pubspec.lock
```
Comment (karel-rehor) [inline .gitignore:8]: So if `pubspec.lock` is being ignored, does this comment still make sense?  My understanding is that `pubspec.lock` gets generated locally based on `pubspec.yaml` and that not every generated `pubspec.lock` will be universally portable.

**Review by karel-rehor — CHANGES_REQUESTED** [review]
Would like to approve, but a comment in `.gitignore` doesn't make sense to me.

File: .gitignore (hunk 1 above)
Comment (bednar) [reply .gitignore:8]: You are right I will remove this comment.

Comment (bednar) [reply .gitignore]: Fixed.

**Review by karel-rehor — APPROVED** [review]
Looks OK to me.

**Pull request merged**